src/main/java/com/google/api/codegen/csharp/CSharpSnippetSetRunner.java

```diff
@@ -48,23 +49,20 @@ public GeneratedResult generate(
             snippetFileName,
             ImmutableMap.<String, Object>of("context", context));
 
-    // TODO don't depend on a cast here
-    CSharpGapicContext csharpContext = (CSharpGapicContext) context;
-
     String outputFilename = snippets.generateFilename(element).prettyPrint();
+    CSharpContextCommon csharpContextCommon = new CSharpContextCommon();
```

Comment:
It's all been fairly significantly changed over the last couple of months. I've essentially just used the Java snippet code and changed it so it does what we need in C#